vast/integration/vast_integration_suite.yaml

```diff
@@ -922,3 +922,48 @@ tests:
         expected_result: error
       - command: exec 'export | top "" | to stdout'
         expected_result: error
+
+  # The summarize operator supports using fields which do not exist, using
+  # `null` instead of their value. Here, we test many combinations of this
+  # behavior. We use the letters A, N and S for all, none and some,
+  # respectively. For example, SA means that the some (but not all)
+  # schemas do not have the aggregation column present, but for all
+  # of them, the group-by column exists.
+  Summarize All None Some:
+    tags: [pipelines]
+    steps:
+      # AA
+      - command: exec 'read zeek-json | summarize x=distinct(_path) by _path'
+        input: data/zeek/zeek.json
+      # NN
+      - command: exec 'read zeek-json | summarize x=distinct(y) by z'
+        input: data/zeek/zeek.json
+      # NA
+      - command: exec 'read zeek-json | summarize x=distinct(y) by _path'
+        input: data/zeek/zeek.json
+      # AN
+      - command: exec 'read zeek-json | summarize x=distinct(_path) by z'
+        input: data/zeek/zeek.json
+      # NS
+      - command: exec 'read zeek-json | summarize x=distinct(y) by id.orig_h'
+        input: data/zeek/zeek.json
+      # SN
+      # TODO: Fix hanging issue after `ctrl.abort(...)` and then enable this test.
+      # - command: exec 'read zeek-json | summarize x=distinct(id.orig_h) by z'
+      #   input: data/zeek/zeek.json
+      #   expected_result: error
+      #   # reason: only one group (`z == null`) and `id.orig_h` conflicts in group
+      # AS
+      - command: exec 'read zeek-json | summarize x=distinct(_path) by id.orig_h'
+        input: data/zeek/zeek.json
+      # SA
+      - command: exec 'read zeek-json | summarize x=distinct(id.orig_h) by _path'
+        input: data/zeek/zeek.json
+      # SS
+      # TODO: This test should succeed, but it fails because the current JSON
+      # parser inference adds a `id.orig_h: ip` with the value `null` to schemas
+      # which should not have it. The bucket `id.orig_h == null` therefore
+      # receives a schema for which `id.orig_h` does not exist, and one for
+      # which it is `null`, which leads to an error.
```

Comment:
@patszt can you add a note to the story that fixes the selector option to re-enable this test?

Reply:
Done

Reply:
Because the final semantics of this are quite forgiving and basically never produce an error, this test passes now. However, I think its output should change after the JSON parser is fixed. I adjusted to note accordingly.